### DISCUSSION TITLE
How to copy a date object?

### DISCUSSION THREAD

**Discussion by github-jing161725** [lodash/lodash#5860] [Q&A]
In the source code, lodash uses "new Date(+dateObj)". Is the '+' necessary?
In MDN, it says that this( new Date(dateObj) ) effectively makes a copy of the existing Date object with the same date and time. 
So, is the '+' necessary?
I tried differernt params, both methods got the same result.
````js
let dateObj=new Date();
let copyDate1=new Date(dateObj);
let copyDate2=new Date(+dateObj);
console.log(copyDate1)
console.log(copyDate2)
```

**Comment by Trott** [comment]
If I were trying to determine the answer to this question, I would remove the `+` from the source code and run the tests to see which (if any) tests fail without the `+`.

**Comment by Trott** [comment]
> In the source code, lodash uses "new Date(+dateObj)".

It does? Where?

**Comment by github-jing161725** [comment]
The link is https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date
It's in the syntax part.
I have tried to use different types of params to create the sourceObj, like timestamps, dateString and ect, and then create the copyObj  in both methods  with and without '+'. The result is the same.

The syntax used to create the sourceObj is list below
```
new Date()
new Date(value)
new Date(dateString)
new Date(dateObject)

new Date(year, monthIndex)
new Date(year, monthIndex, day)
new Date(year, monthIndex, day, hours)
new Date(year, monthIndex, day, hours, minutes)
new Date(year, monthIndex, day, hours, minutes, seconds)
new Date(year, monthIndex, day, hours, minutes, seconds, milliseconds)

Date()

**Reply by Trott** [reply]
Where in the lodash code does it do `new Date(+dateObj)`?

**Reply by github-jing161725** [reply]
_.clone*(clone,cloneDeep,cloneWith,cloneDeepWith) =>baseClone function=>initCloneByTag

**Reply by Trott** [reply]
I'm not sure if this is the motivation or not, but it matters in cases where people are messing about like this:

```js
const myDateObj = new Date()
const fakeDateObj = '0'
fakeDateObj.constructor = myDateObj.constructor

// Note that these are different
console.log(new Date(fakeDateObj))
console.log(new Date(+fakeDateObj))
```
